apps/todoapp/src/app/todos/components/user-info/user-info.component.ts

```diff
@@ -1,11 +1,23 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnChanges } from '@angular/core';
 import { DiscordUser } from '../../types/discordUser.interface';
+import { DiscordService } from '../../services/discord.service';
 
 @Component({
   selector: 'todolist-user-info',
   templateUrl: './user-info.component.html',
   styleUrls: ['./user-info.component.scss'],
 })
-export class UserInfoComponent {
+export class UserInfoComponent implements OnChanges {
   @Input('user') user!: DiscordUser;
+  userAvatar!: string;
+
+  constructor(private discordService: DiscordService) {}
+
+  ngOnChanges() {
```

Comment:
why do we need to load the avatar using ngOnChanges?

Reply:
because in ngOnInit user is undefined

Reply:
yeah because u're accessing the passed user prop, u could load it using ngOnInit using the authService.$user